o1vm/src/interpreters/mips/witness.rs

```diff
@@ -362,12 +362,26 @@ impl<Fp: Field, PreImageOracle: PreImageOracleT> InterpreterEnv for Env<Fp, PreI
     }
 
     fn equal(&mut self, x: &Self::Variable, y: &Self::Variable) -> Self::Variable {
-        // To avoid subtraction overflow in the witness interpreter for u32
-        if x > y {
-            self.is_zero(&(*x - *y))
-        } else {
-            self.is_zero(&(*y - *x))
-        }
+        // We replicate is_zero(x-y), but working on field elt,
+        // to avoid subtraction overflow in the witness interpreter for u32
+        let to_zero_test = Fp::from(*x) - Fp::from(*y);
```

Comment:
You don't need to convert as early as here in `Fp`.

Reply:
Indeed, but I believe this to be more clear, and the performance is negligibly impacted